contracts/contracts/sbtc-withdrawal.clar

```diff
@@ -33,6 +45,123 @@
   )
 )
 
+;; Accept a withdrawal request
+(define-public (accept-withdrawal-request (request-id uint) 
+                                          (bitcoin-txid (buff 32)) 
+                                          (signer-bitmap uint)
+                                          (output-index uint)
+                                          (fee uint))
+  (let 
+    (
+      (current-signer-data (contract-call? .sbtc-registry get-current-signer-data))   
+      (request  (unwrap! (contract-call? .sbtc-registry get-withdrawal-request request-id) ERR_INVALID_REQUEST))
+      (requested-max-fee (get max-fee request))
+      (requester (get sender request))
+    )
+      ;; Check that the caller is the current signer principal
+      (asserts! (is-eq (get current-signer-principal current-signer-data) tx-sender) ERR_INVALID_CALLER)
+
+      ;; Check whether it was already accepted or rejected
+      (asserts! (is-none (get status request)) ERR_ALREADY_PROCESSED)
+
+      ;; Check that fee is not higher than requesters max fee
+      (asserts! (<= fee requested-max-fee) ERR_FEE_TOO_HIGH)
+
+      ;; Burn the locked-sbtc
+      (try! (contract-call? .sbtc-token protocol-burn-locked (get amount request) requester))
+
+      ;; Mint the difference b/w max-fee of the request & fee actually paid back to the user in sBTC
+      (if (is-eq (- requested-max-fee fee) u0)
+        true
+        (try! (contract-call? .sbtc-token protocol-mint (- requested-max-fee fee) requester))
+      )
+
+      ;; Call into registry to confirm accepted withdrawal
+      (try! (contract-call? .sbtc-registry complete-withdrawal request-id true (some bitcoin-txid) (some signer-bitmap) (some output-index) (some fee)))
+
+      (ok true)
+  )
+)
+
+;; Reject a withdrawal request
+(define-public (reject-withdrawal (request-id uint) (signer-bitmap uint))
```

Comment:
Shouldn't the function name be `reject-withdrawal-request`?

Reply:
Ah. Yeap, changed!